dashboard/src/reducers/namespace.ts

```diff
@@ -31,16 +34,19 @@ const namespaceReducer = (
       return { ...state, errorMsg: action.payload.err.message };
     case getType(actions.namespace.clearNamespaces):
       return { ...initialState };
-    case getType(actions.namespace.namespaceReceived):
-      const currentNamespace = state.current === "" ? action.payload : state.current;
-      return { ...initialState, current: currentNamespace };
     case LOCATION_CHANGE:
       const pathname = action.payload.location.pathname;
       // looks for /ns/:namespace in URL
       const matches = pathname.match(/\/ns\/([^/]*)/);
       if (matches) {
         return { ...state, current: matches[1] };
       }
+    case getType(actions.auth.setAuthenticated):
```

Comment:
regarding your Typescript error. The problem is that `LOCATION_CHANGE` doesn't have a `break;` at the end of the case so this one is also executed after the previous one if it matches. That's why you are seeing the error, the `payload` here can be a location change.

To fix it, either add a `break` in the `case LOCATION_CHANGE` or move this case above (but you also need to add a `break;`)

Reply:
Excellent, thanks.